src/ui/SwiftOutputChannel.ts

```diff
@@ -65,4 +102,47 @@ export class SwiftOutputChannel {
             second: "numeric",
         });
     }
+
+    get logs(): string[] {
+        return this.logStore.logs;
+    }
+}
+
+class RollingLog {
+    private _logs: string[] = [];
+    private currentLogLength: number = 0;
+
+    constructor(private maxSizeCharacters: number) {}
+
+    public get logs(): string[] {
+        return [...this._logs];
+    }
+
+    appendLine(log: string) {
+        // It can be costly to calculate the actual memory size of a string in Node so just
+        // use the total number of characters in the logs as a huristic for total size.
+        const logSize = log.length;
+
+        while (this.currentLogLength + logSize > this.maxSizeCharacters && this.logs.length > 0) {
+            const oldestLog = this.logs.shift();
```

Comment:
I don't know how `shift` works internally but this is going to be either re-allocating large buffers of memory, or not freeing up any memory (because the array is equivalent to an array slice). Perhaps we should use a circular buffer. It might be harder to keep a consistent maximum memory usage but wouldn't require creating new arrays of strings repeatedly.

Reply:
I refactored the RollingLog to use a ring buffer style algorithm.